shimmingtoolbox/optimizer/basic_optimizer.py

```diff
@@ -60,23 +61,23 @@ def optimize(self, unshimmed, mask, mask_origin=(0, 0, 0), bounds=None):
             unshimmed (numpy.ndarray): (X, Y, Z) 3d array of unshimmed volume
             mask (numpy.ndarray): (X, Y, Z) 3d array of integers marking volume for optimization -- 0 indicates unused
             mask_origin (tuple): Origin of mask if mask volume does not cover unshimmed volume
-            bounds (list): List of ``(min, max)`` pairs for each coil channels. None
-               is used to specify no bound.
+            bounds (list): List of ``(min, max)`` pairs for each coil channels. UNUSED in pseudo-inverse
         """
         # Check for sizing errors
         self._check_sizing(unshimmed, mask, mask_origin=mask_origin, bounds=bounds)
 
         # Set up output currents and optimize
         output = np.zeros(self.N)
 
-        mx, my, mz = mask_origin
-        mX, mY, mZ = mask.shape
+        mask_range = tuple([slice(mask_origin[i], mask_origin[i] + mask.shape[i]) for i in range(3)])
+        mask_vec = mask.reshape((-1,))
 
         # Simple pseudo-inverse optimization
-        profile_mat = np.reshape(np.transpose(self.coils[mx:mx+mX, my:my+mY, mz:mz+mZ], axes=(3, 0, 1, 2)), (self.N, -1)).T # mV x N
-        unshimmed_vec = np.reshape(unshimmed[mx:mx+mX, my:my+mY, mz:mz+mZ], (mX * mY * mZ,)) # mV
+        # Reshape coil profile: X, Y, Z, N --> [mask.shape], N --> N, [mask.shape] --> N, mask.size --> mask.size, N --> masked points, N
+        coil_mat = np.reshape(np.transpose(self.coils[mask_range], axes=(3, 0, 1, 2)), (self.N, -1)).T[mask_vec != 0, :] # masked points x N
```

Comment:
```suggestion
        coil_mat = np.reshape(np.transpose(self.coils[mask_range], axes=(3, 0, 1, 2)), (self.N, -1)).T[mask_vec != 0, :]  # masked points x N
```
PEP8 wants 2 spaces between code and inline comments